### PR TITLE
Upgrade rabbitmq to v3.8.9

### PR DESCRIPTION
**Pull request by atsikham**
Task #1854: implemented upgrade for RabbitMQ. Latest version is compatible with previous 3.8.3 and 3.7.x.
See https://www.rabbitmq.com/upgrade.html for more details.

### CODE REVIEW

File: core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/reconfigure-rabbitmq-app.yml [hunk 1]
```diff
@@ -7,8 +7,9 @@
   block:
     - name: after-patching | Get rabbitmq namespace
       shell: |-
-        kubectl get statefulsets.apps --all-namespaces -o=jsonpath='{range .items[*]}{.metadata.namespace}{"\t"}{.spec.template.spec.containers[].image}{"\n"}{end}'|
-        awk '/rabbitmq/ {print $1}'
+        kubectl get statefulsets.apps --all-namespaces \
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/reconfigure-rabbitmq-app.yml:10]: @atsikham how about replacing all that ugly code with something like this (:thinking:):
```yaml
- command: kubectl get statefulsets --all-namespaces --output json
  register: shell_get_statefulsets
  changed_when: false

- command: kubectl delete pods --namespace {{ item }} --all
  loop: "{{ _namespaces }}"

  when: _namespaces | length > 0

  vars:
    _namespaces: >-
      {%- set output = [] -%}
      {%- for item in (shell_get_statefulsets.stdout | from_json)["items"] -%}
        {%- for container in item.spec.template.spec.containers -%}
          {%- if '/rabbitmq:' in container.image -%}
            {{- output.append(item.metadata.namespace) -}}
          {%- endif -%}
        {%- endfor -%}
      {%- endfor -%}
      {{- output -}}
```

Comment (atsikham) [reply core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/reconfigure-rabbitmq-app.yml:10]: @sk4zuzu I suggest to create another PR for that as testing scope is different. I changed that line and it's showed here as there was a linter warning.

Comment (sk4zuzu) [reply core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/reconfigure-rabbitmq-app.yml:10]: That's fine! If it gets there somehow I'm :+1::heart_eyes:

File: core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/rabbitmq.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+---
+- name: RabbitMQ | Set rabbitmq_master fact
+  set_fact:
+    rabbitmq_master: "{{ groups['rabbitmq'][0] }}"
+
+- name: RabbitMQ | Check version
+  command: rabbitmqctl version
+  register: rabbitmq_version_current
+
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/rabbitmq.yml:9]:
```suggestion
  changed_when: false

```

**Review by sk4zuzu — CHANGES_REQUESTED** [review]

**Review by sk4zuzu — APPROVED** [review]
LGTM

**Comment by toszo** [comment]
I like the upgrade.md update - it sets a context and makes clear how upgrade of RMQ will look like. 
LGTM

**Review by toszo — APPROVED** [review]

**Pull request merged**